PedroChaparro/loomies-mobile
# Feat: Interact with Loomie / Gyms in map 3D

## What includes

- Triggers an event when the user taps a model in the map.
- Adds simple geometry colliders to the complex models so that the tap check is way more efficient.
- Adds a way to access the navigator without inheriting props.
- Adds an environment variable that defines the distance the user can reach for interacting with map elements.
- When a Loomie is selected and is near enough, the user is redirected to the capture view.

![image](https://user-images.githubusercontent.com/61242172/229918957-30632d88-6ca7-421f-bc16-a1943edcc4ae.png)


## How to test

- In the map view click a Loomie on the map and you should be redirected to the capture page.
- Click a gym on the map and the terminal output should confirm it.

## Notes

- The env variable PLAYER_REACH_RADIUS needs to be synced with the one in the backend.
- There is room for visual improvements, like adding a ring around the player indicating how far he can reach.
